### PR TITLE
Allows external rendering of other filetypes

### LINKED ISSUE #4996: Apply external markup rendering to non-text files
- Gitea version (or commit ref): 1.4.3
- Git version: 2.18.0
- Operating system: FreeBSD 11.2-RELEASE
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [x] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [x] Not relevant
- Log gist:

## Description

Currently, files are only checked against external markup extensions when the file is a text file: https://github.com/go-gitea/gitea/blob/fc0001caa1e9873a86990f7815a75d8f75fbc309/routers/repo/view.go#L201-L214

I would like to be able to use external markup renderers to render previews of some binary formats, similar to how PDFs can be displayed inline. Instead, I always get the "view raw" page for binary files.

### PR DESCRIPTION
**Pull request by HarvsG**
Fixes #4996 and #7614 
allows rendering of non-text files, or otherwise un-accounted for filetypes

### CODE REVIEW

File: routers/repo/view.go [hunk 1]
```diff
@@ -342,7 +342,22 @@ func renderFile(ctx *context.Context, entry *git.TreeEntry, treeLink, rawLink st
 		ctx.Data["IsAudioFile"] = true
 	case base.IsImageFile(buf):
 		ctx.Data["IsImageFile"] = true
+	default:
+		d, _ := ioutil.ReadAll(dataRc)
```
Comment (lafriks) [inline routers/repo/view.go]: It should not be read before max filesize check

**Review by lafriks — CHANGES_REQUESTED** [review]

**Review by HarvsG — COMMENTED** [review]
@lafriks. Agreed, I have made the changes. I wonder if it might also be worth incorporating the fix I have suggested for #8299, to keep this to a single PR.

**Comment by guillep2k** [comment]
I don't understand, why do changes from other PRs show up as changes here? Normally when master is merged into the PR the changes don't show up. Is this an EOL issue?

**Comment by HarvsG** [comment]
@guillep2k It is possible I did a git pull with gitea/gitea still set as my remote? I'm not sure how to revert this without closing the PR and re-opening

**Comment by guillep2k** [comment]
@HarvsG I don't see the problem if you need to open a new PR. Just reference it here (you need of course to create another branch).

**Comment by guillep2k** [comment]
You can cherry-pick your own commits, I guess.

**Comment by guillep2k** [comment]
You should close this PR.

**Pull request closed without merging**